tests/trainer/test_trainer.py

```diff
@@ -461,6 +511,55 @@ def on_load_checkpoint(self, _):
         assert next_model.num_on_load_checkpoint_called == 1
 
 
+def test_trainer_properties_resume_from_checkpoint(tmpdir):
```

Comment:
not a fan of this test because it's relying on the callback system. we could more efficiently test with unit testing by doing something like this:

```python
checkpoint_connector = CheckpointConnector(trainer, ...) 
trainer.state.fn = FIT
checkpoint.connector.resume_start()
checkpoint_connector.restore_xyz()
assert trainer.current_epoch == ...
```


this has several benefits, but most importantly it makes it very clear we are testing the checkpoint connector and not the trainer (which has no business in restoring anything).

Reply:
yes, I can do that.. but still need optimizers, schedulers, and lighting module to be setup correctly which can be done withing trainer.fit/validate/test/ 
also, there are few calls that happen during `.fit` only which are required to be tested too. callback seems to be an ideal way to do this end-to-end testing.. ensuring the required functions are called correctly and required things are restored correctly.